File: tests/src/main/java/com/github/klee0kai/test/lifecycle/structure/DataStorage.java
package com.github.klee0kai.test.lifecycle.structure;

import com.github.klee0kai.test.lifecycle.di.qualifiers.DataStorageSize;

import java.util.UUID;

public class DataStorage {

    public final UUID uuid = UUID.randomUUID();

    public final String size;

    public DataStorage() {
        size = null;
    }

    public DataStorage(DataStorageSize size) {
        this.size = size.size;
    }

}

File: test_kotlin/src/main/kotlin/com/github/klee0kai/test/lifecycle/structure/DataStorage.kt
package com.github.klee0kai.test.lifecycle.structure

import com.github.klee0kai.test.lifecycle.di.qualifier.DataStorageSize
import java.util.*

class DataStorage {
    val uuid = UUID.randomUUID()
    val size: String?

    constructor() {
        size = null
    }

    constructor(size: DataStorageSize) {
        this.size = size.size
    }
}